packages/types/src/sign-client/auth.ts

```diff
@@ -0,0 +1,174 @@
+import {
+  ErrorResponse,
+  JsonRpcError,
+  JsonRpcRequest,
+  JsonRpcResponse,
+  JsonRpcResult,
+} from "@walletconnect/jsonrpc-types";
+import { CoreTypes, ICore, IStore, Verify } from "../core";
+import { SessionTypes } from "./session";
+
+export declare namespace AuthTypes {
+  type Event = "session_authenticate";
+
+  interface AuthRequestEventArgs {
+    requester: Participant;
+    authPayload: PayloadParams;
+    expiryTimestamp: number;
+  }
+
+  type AuthResponseEventArgs =
+    | { message: string; code: number }
+    | JsonRpcResult<Cacao>
+    | JsonRpcError;
+
+  interface BaseEventArgs<T = unknown> {
+    id: number;
+    topic: string;
+    params: T;
+    verifyContext?: Verify.Context;
+  }
+
+  interface EventArguments {
+    auth_request: BaseEventArgs<AuthRequestEventArgs>;
+    auth_response: BaseEventArgs<AuthResponseEventArgs>;
+    sign_request: BaseEventArgs<{
+      request: { method: string; params: any };
+      chainId: string;
+    }>;
+    sign_response: BaseEventArgs<JsonRpcResult | JsonRpcError>;
+  }
+
+  interface Options extends CoreTypes.Options {
+    metadata: Metadata;
+    core?: ICore;
+    projectId: string;
+  }
+
+  interface Metadata {
+    name: string;
+    description: string;
+    url: string;
+    icons: string[];
+    redirect?: {
+      native?: string;
+      universal?: string;
+    };
+    verifyUrl?: string;
+  }
+
+  interface EventCallback<T extends JsonRpcRequest | JsonRpcResponse> {
+    topic: string;
+    payload: T;
+  }
+
+  /**
+   * `aud` is used in protocol request
+   * `uri` is more descriptive and is used in client APIs
+   * formatMessageParams should accept either `aud` or `uri` as a parameter to construct the message
+   */
+  type FormatMessageParams = {
+    aud?: string;
+    uri?: string;
+  } & Omit<BaseAuthRequestParams, "aud" | "chainId">;
+
+  interface BaseAuthRequestParams {
+    domain: string;
+    aud: string;
+    nonce: string;
+    version?: string;
+    iat?: string;
+    nbf?: string;
+    exp?: string;
+    chainId?: string;
+    statement?: string;
+    requestId?: string;
+    resources?: string[];
+    expiry?: number;
```

Comment:
Not sure if a decision was taken on this but the [specs discussion](https://github.com/WalletConnect/walletconnect-specs/pull/171/files#r1432667513) suggests to rename `expiry` to `ttl`
```suggestion
    ttl?: number;
```

Reply:
`expiry` the public API, `ttl` is derived from the `expiry` devs are setting